src/state.rs

```diff
@@ -58,6 +58,8 @@ impl LeaderState {
         self.next_index.insert(follower, index);
     }
 
+    // TODO: why is this never used?
+    #[allow(dead_code)]
```

Comment:
I've checked it out and looks like the code is safe to remove since only `count_match_indexes` is actually called. Removing.